### PR TITLE
implement create launcher shortcut

### PR DESCRIPTION
**Pull request by chizoba**
fix for issue #170

### CODE REVIEW

**Comment by gsantner** [comment]
Thanks! But you didn't read the issue fully. The target is not to start the app (via action.main), but to start the app with an specific specified URL. The one that is currently open in the WebView, when creating the shortcut.

**Comment by chizoba** [comment]
okay, let me do that

**Comment by vanitasvitae** [comment]
I'd propose that instead of using the pod domain as icon title, a dialog is shown asking the user for a name for the shortcut. You could then prefill the EditText with either the pod domain, or even better the name of the site/post that's currently loaded.

**Comment by gsantner** [comment]
Thats intended by the issue too :D

**Comment by chizoba** [comment]
@gsantner you haven't reviewed the last commit I made...

**Comment by gsantner** [comment]
sorry, didn't know if you finished

File: app/src/main/java/com/github/dfa/diaspora_android/activity/MainActivity.java [hunk 1]
```diff
@@ -1161,11 +1161,12 @@ public boolean onNavigationItemSelected(MenuItem item) {
...
             case R.id.nav_shortcut: {
                 Intent shortcutIntent = new Intent(MainActivity.this, MainActivity.class);
-                shortcutIntent.setAction(Intent.ACTION_MAIN);
+                shortcutIntent.setAction(Intent.ACTION_VIEW);
+                shortcutIntent.setData(Uri.parse(urls.getStreamUrl()));
                 // The intent responsible for creating the shortcut
                 Intent intent = new Intent();
                 intent.putExtra(Intent.EXTRA_SHORTCUT_INTENT, shortcutIntent);
-                intent.putExtra(Intent.EXTRA_SHORTCUT_NAME, getString(R.string.app_name));
+                intent.putExtra(Intent.EXTRA_SHORTCUT_NAME, _appSettings.getPod().getName());
```
Comment (gsantner) [inline app/src/main/java/com/github/dfa/diaspora_android/activity/MainActivity.java]: extract title from current webview's title

File: app/src/main/java/com/github/dfa/diaspora_android/activity/MainActivity.java [hunk 2]
```diff
@@ -1161,11 +1161,12 @@ public boolean onNavigationItemSelected(MenuItem item) {
...
                 intent.putExtra(Intent.EXTRA_SHORTCUT_ICON_RESOURCE, Intent.ShortcutIconResource.fromContext(MainActivity.this, R.drawable.ic_launcher));
```
Comment (gsantner) [inline app/src/main/java/com/github/dfa/diaspora_android/activity/MainActivity.java]: Extract icon from current webviews favicon. If there is non, fallback to launcher icon (which you currrently using)

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -7,6 +7,8 @@
     <uses-permission android:name="android.permission.INTERNET" />
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
+    // Permission to install shortcut icon
```
Comment (gsantner) [inline app/src/main/AndroidManifest.xml:10]: Remove or use xml comment: `<!-- comment -->`

**Comment by gsantner** [comment]
@chizoba Sorry didnt know you finished. I added review

**Comment by vanitasvitae** [comment]
@chizoba Are you still working on this, do you need feedback or anything?

**Comment by chizoba** [comment]
@vanitasvitae yes i am. i have just been busy lately. 
Quick question, to be able to get the web view's favicon, i need to override onReceivedIcon() method of the respective WebChromeClient right?

**Comment by vanitasvitae** [comment]
@chizoba Possibly. I didn't test it though.

**Comment by gsantner** [comment]
Closing as this seems not get resolved :). For reference if somebody want to work on this, I did it in another app. See this commit, which has all needed. <https://github.com/gsantner/markor/commit/a3fd314c46839fb958a87307d5e2c91fb95b161d#diff-104394ddb27792db13d06de2874f85c7R45>

**Pull request closed without merging**